### ISSUE TITLE
Module build failed: Error: Emit skipped at State.emit

### ISSUE THREAD

**Issue by EvanWillms** [s-panferov/awesome-typescript-loader#101]
Anytime I try to build an Ionic 2.0.0-beta.8 app with webpack and awesome-typescript-loader that had previously successfully built under babel-loader, I get an error of "Module build failed: Error: Emit skipped".  My webpack.config.js and tsconfig.json files should be good based on observed consistency across several sample projects.

```
Module build failed: Error: Emit skipped
    at State.emit (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/dist.babel/host.js:253:23)
    at transform (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/dist.babel/index.js:168:48)
    at _callee2$ (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/dist.babel/index.js:218:42)
    at tryCatch (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/babel-polyfill/node_modules/babel-regenerator-runtime/runtime.js:61:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/babel-polyfill/node_modules/babel-regenerator-runtime/runtime.js:329:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/babel-polyfill/node_modules/babel-regenerator-runtime/runtime.js:94:21)
    at step (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/dist.babel/index.js:30:41)
    at onfulfill (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/dist.babel/index.js:17:17)
    at tryCatcher (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/bluebird/js/main/promise.js:507:31)
    at Promise._settlePromiseAt (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/bluebird/js/main/promise.js:581:18)
    at Promise._settlePromises (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/bluebird/js/main/promise.js:697:14)
    at Async._drainQueue (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/evan/Downloads/9144OS-Code/Code bundles/s1_end/node_modules/awesome-typescript-loader/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
 @ multi main (CLI v2.0.0-beta.17)

```

Full npm install and run:
https://gist.github.com/EvanWillms/8e691c5a9348d5934ac6

webpack.config.js:
https://gist.github.com/EvanWillms/c55764139f66b0db3640

tsconfig.json:
https://gist.github.com/EvanWillms/c0638158f2c7c8853a64

Any ideas on what I've done wrong?

**Comment by s-panferov** [comment]
@EvanWillms 	I will take a look today

**Comment by s-panferov** [comment]
@EvanWillms 	JS compilation is a quite new TS feature and there are a lot of issues right now. It looks like that the problem is in TypeScript itself right now https://github.com/Microsoft/TypeScript/issues/7363

**Comment by s-panferov** [comment]
@EvanWillms JS support added in version v0.16.0-rc.0. Please note that support is **experimental** for now and a lot of testing is required. Could you please check this version on your codebase?

**Comment by jasdeepsingh** [comment]
I'm having this exact same issue, I'll try out the v0.16.0-rc.0 @s-panferov

**Comment by s-panferov** [comment]
@jasdeepsingh @EvanWillms 	

could you please tell me about results?

**Comment by s-panferov** [comment]
Landed in v0.16.0
